### PR TITLE
Add aws-creds flag to hypershift-operator

### PR DESCRIPTION
**Pull request by sjenning**
More AWS Private cluster support work.

It adds an optional `--aws-creds` flag to `hypershift install`.

The `hypershift-operator` will need AWS credentials sufficient to create Endpoint Services in AWS.

These creds will be plumbed into the controller that creates the Endpoint Service in AWS in response to `AWSEndpointService` creation in the next PR.

If you don't pass `aws-creds`, the `hypershift-operator-creds` secret `credentials` key is just empty.  We can then use this as an indicator on whether or not to start the `AWSEndpointService` controller in the `hypershift-operator`.  If the `credentials` key is empty, don't start the controller.

### CODE REVIEW

File: cmd/install/install.go [hunk 1]
```diff
@@ -67,6 +69,7 @@ func NewCommand() *cobra.Command {
 	cmd.Flags().BoolVar(&opts.ExcludeEtcdManifests, "exclude-etcd", false, "Leave out etcd manifests")
 	cmd.Flags().BoolVar(&opts.EnableOCPClusterMonitoring, "enable-ocp-cluster-monitoring", opts.EnableOCPClusterMonitoring, "Development-only option that will make your OCP cluster unsupported: If the cluster Prometheus should be configured to scrape metrics")
 	cmd.Flags().BoolVar(&opts.EnableCIDebugOutput, "enable-ci-debug-output", opts.EnableCIDebugOutput, "If extra CI debug output should be enabled")
+	cmd.Flags().StringVar(&opts.AWSCreds, "aws-creds", opts.AWSCreds, "Path to an AWS credentials file (AWS only)")
```
Comment (ironcladlou) [inline cmd/install/install.go:72]: If this is now required, maybe [the docs](https://hypershift-docs.netlify.app/quickstart/#create-a-cluster) should be updated to reflect the new requirement? 😁

Comment (ironcladlou) [reply cmd/install/install.go:72]: I see that it's not required here... when _is_ it required?

Comment (alvaroaleman) [reply cmd/install/install.go:72]: Shouldn't this default to then value of `AWS_SHARED_CREDENTIALS_FILE`?

Comment (sjenning) [reply cmd/install/install.go:72]: @ironcladlou it is not required.  Updated the PR description better explaining this.

@alvaroaleman a good idea, but a little too magical I think.  The user might have an over privileged cred set in their environment and we don't want use that implicitly IMHO.

**Review by alvaroaleman — APPROVED** [review]

File: cmd/install/install.go (hunk 1 above)
Comment (ironcladlou) [reply cmd/install/install.go:72]: >The user might have an over privileged cred set in their environment and we don't want use that implicitly IMHO.

Good point, I think this convinces me explicit is safer

**Pull request merged**